Dockerfile.slim

```diff
@@ -0,0 +1,60 @@
+FROM clojure:lein-alpine
+RUN apk add --no-cache make
+# Install only dependencies
+WORKDIR /app
+COPY project.clj .
+COPY resources/puppetlabs/puppetdb/bootstrap.cfg \
+    ./resources/puppetlabs/puppetdb/bootstrap.cfg
+RUN lein with-profile uberjar deps
+# Build uberjar -- see .dockerignore
+COPY . .
+RUN lein with-profile uberjar uberjar
+
+
+FROM openjdk:8-jre-alpine
+RUN apk add --no-cache tini curl openssl
+COPY --from=0 /app/target/puppetdb.jar /
+ENV PUPPETDB_DATABASE_CONNECTION="//postgres:5432/puppetdb"
+ENV PUPPETDB_USER=puppetdb
+ENV PUPPETDB_PASSWORD=puppetdb
+ENV PUPPETDB_NODE_TTL=7d
+ENV PUPPETDB_NODE_PURGE_TTL=14d
+ENV PUPPETDB_REPORT_TTL=14d
+ENV PUPPETDB_JAVA_ARGS="-Djava.net.preferIPv4Stack=true -Xms256m -Xmx256m"
+ENV USE_PUPPETSERVER=true
+
+# Values from /etc/default/puppetdb
+ENV JAVA_BIN="/usr/bin/java"
+ENV JAVA_ARGS="$PUPPETDB_JAVA_ARGS"
+ENV USER="puppetdb"
+ENV GROUP="puppetdb"
+ENV INSTALL_DIR="/opt/puppetlabs/server/apps/puppetdb"
+ENV CONFIG="/etc/puppetlabs/puppetdb/conf.d"
+ENV BOOTSTRAP_CONFIG="/etc/puppetlabs/puppetdb/bootstrap.cfg"
+ENV SERVICE_STOP_RETRIES=60
+
+COPY docker/puppetdb/logging /etc/puppetlabs/puppetdb/logging/
+COPY docker/puppetdb/conf.d /etc/puppetlabs/puppetdb/conf.d/
+COPY resources/puppetlabs/puppetdb/bootstrap.cfg /etc/puppetlabs/puppetdb/
+COPY resources/ext/config/logback.xml /etc/puppetlabs/puppetdb/
+COPY resources/ext/config/request-logging.xml /etc/puppetlabs/puppetdb/
+RUN mkdir -p /opt/puppetlabs/server/data/puppetdb
+
+RUN addgroup $GROUP && adduser -S $USER -G $GROUP
+
+ADD https://raw.githubusercontent.com/puppetlabs/pupperware/puppetless-ssl/shared/ssl.sh /ssl.sh
```

Comment:
We will change this URL from `puppetless-ssl` (the branch name) to a proper SHA so we "pin" the dependency.